frontend/src/pages/search/Search.jsx

```diff
@@ -109,9 +116,23 @@ function Search() {
                     />
                   </div>
                 ))
-              ) : ( <p>No products found</p>)}
+              ) : !loader && (
+                <p>No products found</p>
+              )}
+            </div>
+            <div className="search-loader-div">
+            <MagnifyingGlass
+              visible={loader}
+              height="121"
+              width="121"
+              ariaLabel="MagnifyingGlass-loading"
+              wrapperStyle={{}}
+              wrapperClass="MagnifyingGlass-wrapper"
+              glassColor="#ECECEC"
+              color="#8367D8"
+            />
```

Comment:
Wrap this in a custom `Loader` component so it can be easily changed in all places later on.